samza-core/src/main/java/org/apache/samza/operators/StreamGraphImpl.java

```diff
@@ -231,9 +241,8 @@ public MessageStreamImpl getInputStream(SystemStream sstream) {
    * @param <M>  the type of input message
    * @return  the {@link OutputStream} object for the re-partitioned stream
    */
-  <PK, M> MessageStreamImpl<M> createIntStream(Function<M, PK> parKeyFn) {
-    // TODO: placeholder to auto-generate intermediate streams via {@link StreamSpec}
-    StreamSpec streamSpec = this.createIntStreamSpec();
+  <PK, M> MessageStreamImpl<M> createIntStream(String streamId, Function<M, PK> parKeyFn) {
+    StreamSpec streamSpec = runner.streamFromConfig(streamId);
```

Comment:
I have the same question. I think we will need a util method to create the intermediate StreamSpec from the StreamGraphImpl, partitionBy operator, and application ID (I would argue this would have the same meaning as the Pipeline ID that Jacob came up w/ earlier).